core/domain/state_domain.py

```diff
@@ -745,16 +758,336 @@ def get_all_html_content_strings(self):
+        InteractionCustomizationArgument domain object. This is done by
+        traversing the customization argument schema, and converting
+        unicode to SubtitledUnicode and html to SubtitledHtml where appropriate.
+
+        Args:
+            ca_dict: dict. The customization argument dictionary. A dict of the
+                single key 'value' to the value of the customization argument.
+            ca_schema: dict. The schema that defines the customization argument
+                value.
+
+        Returns:
+            InteractionCustomizationArg. The customization argument domain
+            object.
+        """
+        def convert_content_to_domain_obj(ca_value, obj_type):
+            """Conversion function used to convert SubtitledHtml dicts to
+            SubtitledHtml and SubtitledUnicode dicts to SubtitledUnicode.
+
+            Args:
+                ca_value: dict. Dictionary of key 'value' to
+                    original value of customization argument.
+                obj_type: str. Indicates the obj_type found in
+                    the customization arguments schema.
+
+            Returns:
+                dict. The unmodified customization argument value.
+            """
+            if obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_UNICODE:
+                return SubtitledUnicode(
+                    ca_value['content_id'], ca_value['unicode_str'])
+
+            if obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_HTML:
+                return SubtitledHtml(
+                    ca_value['content_id'], ca_value['html'])
+
+        ca_value = InteractionCustomizationArg.traverse_by_schema_and_convert(
+            ca_schema,
+            copy.deepcopy(ca_dict['value']),
+            convert_content_to_domain_obj
+        )
+
+        return cls(ca_value, ca_schema)
+
+    def __init__(self, value, schema):
+        """Initializes a InteractionCustomizationArg domain object.
+
+        Args:
+            value: *. The value of the interaction customization argument.
+            schema: dict. The schema defining the specification of the value.
+        """
+        self.value = value
+        self.schema = schema
+
+    def get_content_ids(self):
+        """Get all content_id's from SubtitledHtml and SubtitledUnicode in the
+        customization argument.
+
+        Returns:
+            list(str). A list of content_id's.
+        """
+        return InteractionCustomizationArg.traverse_by_schema_and_get(
+            self.schema,
+            self.value,
+            ('SubtitledHtml', 'SubtitledUnicode'),
+            lambda x: x.content_id
+        )
+
+    def get_html(self):
+        """Get all html from SubtitledHtml in the customization argument.
+
+        Returns:
+            list(str). A list of html.
+        """
+
+        return InteractionCustomizationArg.traverse_by_schema_and_get(
+            self.schema,
+            self.value,
+            ('SubtitledHtml'),
```

Comment:
Does this pass tests? It will be regarded as a string, not a 1-tuple...

If it does pass tests, your tests are too weak.